### PR TITLE
Improve calibration

### PR DESCRIPTION
**Pull request by yrenier**
PR #331 has been cancelled, but there has been several proposed and implemented improvements that I belive should not be lost, so this new PR is to include them.

### CODE REVIEW

File: lstchain/calib/camera/calibrator.py [hunk 1]
```diff
@@ -29,13 +29,11 @@ class LSTCameraCalibrator(CameraCalibrator):
 
     calibration_path = Unicode(
         '',
-        allow_none = True,
```
Comment (rlopezcoto) [inline lstchain/calib/camera/calibrator.py:32]: why is this necessary here?

File: lstchain/calib/camera/calibrator.py [hunk 2]
```diff
@@ -130,28 +127,27 @@ def _initialize_correction(self):
                     table = '/tel_' + str(telid) + '/pixel_status'
                     next(h5_table.read(table, self.mon_data.tel[telid].pixel_status))
         except:
-            self.log.error(f"Problem in reading calibration file {self.calibration_path}")
+            self.log.exception(f"Problem in reading calibration file {self.calibration_path}")
+            raise FileNotFoundError(f"Problem in reading calibration file {self.calibration_path}")
```
Comment (rlopezcoto) [inline lstchain/calib/camera/calibrator.py:131]: why this duplication of errors?

File: lstchain/calib/camera/calibrator.py [hunk 3]
```diff
@@ -114,7 +112,6 @@ def _initialize_correction(self):
 
         try:
             with HDF5TableReader(self.calibration_path) as h5_table:
-                assert h5_table._h5file.isopen == True
```
Comment (rlopezcoto) [inline lstchain/calib/camera/calibrator.py:117]: what is the problem on checking here whether the files is opened?

File: setup.py [hunk 4]
```diff
@@ -50,6 +50,7 @@ def find_scripts(script_dir, prefix):
         'scipy',
         'seaborn',
         'tables',
+        'joblib',
```
Comment (rlopezcoto) [inline setup.py:53]: `joblib` is included in the ctapipe environment installation, I do not think we need it here, do we?

File: lstchain/reco/dl1_to_dl2.py [hunk 5]
```diff
@@ -10,8 +10,8 @@
 import numpy as np
 import pandas as pd
 from sklearn.ensemble import RandomForestClassifier, RandomForestRegressor
-from sklearn.externals import joblib
 from sklearn.model_selection import train_test_split
+import joblib
```
Comment (rlopezcoto) [inline lstchain/reco/dl1_to_dl2.py:14]: very good, because this one will get deprecated in `sklearn`

File: lstchain/calib/camera/calibrator.py (hunk 1 above)
Comment (yrenier) [reply lstchain/calib/camera/calibrator.py:32]: the modification came from a comment from @MaxNoe ( https://github.com/cta-observatory/cta-lstchain/pull/331#discussion_r405317641)
Also I could found nowhere in the code where the None value was used.

File: lstchain/calib/camera/calibrator.py (hunk 2 above)
Comment (yrenier) [reply lstchain/calib/camera/calibrator.py:131]: that is based on the comment here: https://github.com/cta-observatory/cta-lstchain/pull/331#discussion_r405313711

File: lstchain/calib/camera/calibrator.py (hunk 3 above)
Comment (yrenier) [reply lstchain/calib/camera/calibrator.py:117]: that check is already performed by pytable, so it is redondant

File: setup.py (hunk 4 above)
Comment (yrenier) [reply setup.py:53]: Rigth, i did not notice that

**Comment by rlopezcoto** [comment]
@FrancaCassol did you have time to have a look at this? There are some functionalities that would certainly be useful

File: setup.py [hunk 6]
```diff
@@ -50,7 +50,6 @@ def find_scripts(script_dir, prefix):
         'scipy',
         'seaborn',
         'tables',
-        'joblib',
```
Comment (MaxNoe) [inline setup.py]: NO! please list everything, that you directly import here. You should not rely on transitive dependencies from ctapipe.

If you have `import package` in lstchain, <package> belongs into the setup.py

File: setup.py (hunk 4 above)
Comment (MaxNoe) [reply setup.py]: No, this is not how this works.

Please list all *direct* dependencies of lstchain in `install_requires`. Do not rely on transitive depdendencies through your dependencies.

Comment (MaxNoe) [reply setup.py]: If you import a package, it goes into `install_requires`

Comment (yrenier) [reply setup.py]: That makes sense, I will add it back

**Comment by FrancaCassol** [comment]
Hi @yrenier 
could you please merge this PR with the present master?

**Comment by rlopezcoto** [comment]
@yrenier can you solve the conflicts of the PR, please?

**Comment by yrenier** [comment]
master is merged, sorry for the delay.

**Comment by rlopezcoto** [comment]
no problem @yrenier and thanks! 
@FrancaCassol will have a look, but these days are a bit busy... this will probably go better to a subsequent major/bugfix release

**Comment by yrenier** [comment]
`but these days are a bit busy... this will probably go better to a subsequent major/bugfix release`
No problem, specially as there is nothing really urgent in this PR.

File: lstchain/calib/camera/calibrator.py [hunk 7]
```diff
@@ -143,14 +141,15 @@ def _calibrate_dl0(self, event, telid):
         create dl0 level, for the moment copy the r1
         """
         waveforms = event.r1.tel[telid].waveform
+        tel_calibration = self.mon_data.tel[telid].calibration
```
Comment (FrancaCassol) [inline lstchain/calib/camera/calibrator.py:144]: Hi @yrenier, 
since in this module we play with different  types of calibration data (self and event), I prefer not to introduce aliases in order to keep the formula immediately understandable.

Comment (yrenier) [reply lstchain/calib/camera/calibrator.py:144]: I did this for readability of the formulas and for performance optimization (probably negligible) . 
I can revert it or change the variable name to `self_tel_calib` for example. Let me know what you prefer.

Comment (rlopezcoto) [reply lstchain/calib/camera/calibrator.py:144]: @yrenier could you please revert it? afterwards, we can merge this PR

Comment (yrenier) [reply lstchain/calib/camera/calibrator.py]: done

Comment (rlopezcoto) [reply lstchain/calib/camera/calibrator.py]: @FrancaCassol are you happy now with this PR?
